### DISCUSSION TITLE
Form reset, append adds ghost empty fields

### DISCUSSION THREAD

**Discussion by imprakharshukla** [react-hook-form/react-hook-form#11898] [Q&A] [answered]
### Version Number

7.51.3

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-forms-appenderror-97c7km?file=%2Fsrc%2FApp.tsx%3A813%2C28

### Steps to reproduce

1. Click on "Add Inventory" button to append a field after the initial data has been set.
2. Click on "Update Product" and observe the console log.
The first click logs an empty error object and then the next click suddenly adds about 12 fields in the inventory key.

### Expected behaviour

The extra 12 ghost empty inventory fields should not be added.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Relevant log output

![image](https://github.com/react-hook-form/react-hook-form/assets/39938009/e553ab33-ddb8-4a9e-b959-fb237e4e05f8)

```shell
[
    null,
    {
        "variant": {
            "message": "Array must contain at least 1 element(s)",
            "type": "too_small"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    },
    {
        "size": {
            "message": "Required",
            "type": "invalid_type"
        },
        "variant": {
            "message": "Required",
            "type": "invalid_type"
        }
    }
]
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Rudomitori** [comment]
I faced a bug that looks similar to the described here.

I've tried different versions of react-hook-form and found that the bug was introduced in version `7.38.0`

**Comment by imprakharshukla** [comment]
@Rudomitori Just tested versions till `7.32.0` and the ghost fields are still created. Makes me wonder if there's anything wrong with the way I am initialising the form (though I have tired to do it in multiple ways and it still happens).

**Comment by bluebill1049** [comment]
clean up your codesandbox.

**Comment by Rudomitori** [comment]
It took some time but I finally managed to reproduce my use case. 
[Codesandbox](https://codesandbox.io/p/sandbox/react-hook-form-usefieldarray-forked-dvzcd8)

Steps:
1. Click "Append"
2. Enter something into the array fields
3. Click "Reset"
   The array fields disappears
4. Click "Append"
5. Enter something into the array fields
6. Click "Reset"
   Expected: Result is the same as after the step 3
   Actual: The array fields are still there, the first field isn't changed, the second field becomes empty

**Comment by erashu212** [comment]
@imprakharshukla  You can fix this issue by passing
This is happening because of this [line](https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/createFormControl.ts#L1212)
```tsx
 const onReset = () => {
    reset({}, { keepDefaultValues: true });
  };
```

**Comment by imprakharshukla** [comment, marked as answer]
@erashu212 I am not sure about that fix but it did stop happening on ver. `7.44.2`
